src/io/SPI.cpp

```diff
@@ -23,48 +27,93 @@ SPI::SPI(GPIO* CSPins[], uint8_t pinLength, Pin sckPin, Pin mosiPin) : CSPinsLen
     }
 }
 
-void SPI::write(uint8_t* bytes, uint8_t length) {
-    for (int i = 0; i < length; i++)
-        write(bytes[i]);
+SPI::SPIStatus SPI::write(uint8_t* bytes, uint8_t length) {
+    SPIStatus status = SPIStatus::OK;
+
+    for (int i = 0; i < length; i++) {
+        status = write(bytes[i]);
```

Comment:
With the previous change, this line would change like this:
```suggestion
        SPI_CHECK(write(bytes[i]));
```